metaflow/plugins/aws/step_functions/step_functions.py

```diff
@@ -328,8 +332,62 @@ def _visit(node, workflow, exit_node=None):
                         )
                     )
                     .max_concurrency(self.max_workers)
-                    .output_path("$.[0]")
+                    .output_path("$" if self.use_distributed_map else "$.[0]")
```

Comment:
This allows us to get the S3 result writer destination when using the distributed map mode.